cachi2/core/package_managers/pip.py

```diff
@@ -1601,52 +1632,153 @@ def _validate_provided_hashes(requirements: list[PipRequirement], require_hashes
                 raise PackageRejected(msg, solution=None)
 
 
-def _download_pypi_package(
-    requirement: PipRequirement,
-    pip_deps_dir: RootedPath,
-    pypi_url: str,
-    pypi_auth: Optional[AuthBase] = None,
-) -> dict[str, Any]:
+@dataclass
+class DistributionPackageInfo:
+    """
+    A class representing relevant information about a distribution package.
+
+    Attributes:
+        name (str): The name of the package.
+        version (str): The version of the package.
+        package_type (Literal): sdist or wheel.
+        path (Path): The path to the package file.
+        url (str): The URL of the package.
+        is_yanked (bool): Indicates whether the package is yanked.
+        pypi_checksums (set[ChecksumInfo]): Set of checksums reported by PyPI.
+        user_checksums (set[ChecksumInfo]): Set of checksums specified by the user.
+        checksums_to_verify (set[ChecksumInfo]): Set of checksums to verify.
     """
-    Download the sdist (source distribution) of a PyPI package.
 
-    The package must be pinned to an exact version using the '==' (or '===') operator.
-    While the specification defines the '==' operator as slightly magical (reference:
-    https://www.python.org/dev/peps/pep-0440/#version-matching), we treat the version
-    as exact (after normalization).
+    name: str
+    version: str
+    package_type: Literal["sdist", "wheel"]
+    path: Path
+    url: str
+    is_yanked: bool
 
-    Does not download any dependencies (implied: ignores extras). Ignores environment
-    markers (target environment is not known to Cachi2).
+    pypi_checksums: set[ChecksumInfo] = field(default_factory=set)
+    user_checksums: set[ChecksumInfo] = field(default_factory=set)
 
-    :param PipRequirement requirement: PyPI requirement from a requirement.txt file
-    :param RootedPath pip_deps_dir: The deps/pip directory in a Cachi2 request bundle
-    :param str pypi_url: URL of the PyPI server or a proxy
-    :param (requests.auth.AuthBase | None) pypi_auth: Authorization for the PyPI server
+    checksums_to_verify: set[ChecksumInfo] = field(init=False, default_factory=set)
 
-    :return: Dict with package name, version and download path
-    :raises FetchError: if PyPI query failed
-    :raises PackageRejected: if sdists for the package is not found or yanked
-    """
-    timeout = get_config().requests_timeout
-    package = requirement.package
+    def __post_init__(self) -> None:
+        if self.package_type == "wheel":
+            self.checksums_to_verify = self.determine_checksums_to_verify(
+                self.pypi_checksums, self.user_checksums
+            )
+
+    def determine_checksums_to_verify(
```

Comment:
I'd make this function private, since it is intended to only be used internally by this class.